src/ast/json.rs

```diff
@@ -401,16 +433,18 @@ impl<'arena> Ast<'arena> for Json<'arena> {
             // fields must have their left hand side be a string
             Json::Field(_) => {
                 if index == 0 {
-                    c == CHAR_STRING
+                    node_type == Self::Class::Str
                 } else {
                     true
                 }
             }
         }
     }
 
-    fn is_valid_root(&self, c: char) -> bool {
-        Self::valid_chars().any(|x| x == c)
+    fn is_valid_root(&self, node_type: Self::Class) -> bool {
```

Comment:
Here, I think it's OK to say `Self::Class` to be consistent with the `Ast` trait's definition, but I wouldn't complain if it's made shorter.